### PR TITLE
Implement map list search

### PR DESCRIPTION
**Pull request by JaThePlayer**
You can now search for maps in the everest map list by hitting the new "Search" button. After you press it, start typing in the name of the map you want to play using the keyboard. Once you're done, press Enter. Useful for when you have tons of maps installed.

### CODE REVIEW

File: Celeste.Mod.mm/Content/Dialog/English.txt [hunk 1]
```diff
@@ -33,6 +33,9 @@
 	MENU_MODOPTIONS_MULTIPLE_MODS_FAILEDTOLOAD=	{0} mods failed to load
 	MENU_MODOPTIONS_UPDATE_AVAILABLE=			An Everest update is available
 
+# File Select Naming
+	NAME_HINT_KEYBOARDINPUT=You're in keyboard input mode.
+	NAME_HINT_USECONTROLLER=Press A on a controller to switch to vanilla mode.
```
Comment (0x0ade) [inline Celeste.Mod.mm/Content/Dialog/English.txt]: Please remove these lines if they're unnecessary for this PR.

File: Celeste.Mod.mm/Content/Dialog/English.txt [hunk 2]
```diff
@@ -108,6 +112,9 @@
 	MODUPDATECHECKER_INSTALLING=	installing...
 	MODUPDATECHECKER_FAILED=		update failed!
 	MODUPDATECHECKER_WILLRESTART=	press Back to restart Celeste
+	MODUPDATECHECKER_UPDATE_ALL=	Update all mods
+	MODUPDATECHECKER_UPDATE_ALL_INPROGRESS=	Updating all mods...
+	MODUPDATECHECKER_UPDATE_ALL_DONE=	All mods have been updated.
```
Comment (0x0ade) [inline Celeste.Mod.mm/Content/Dialog/English.txt]: Uh... this smells like a merge conflict that GitHub is too lazy to spot? We've already got those lines on the master branch.

Please DON'T remove them, as I'm not sure if git will then try to merge the "removal".

**Comment by 0x0ade** [comment]
Other than the dialog mishap, looks good to me unless I'm blind 👍 good job!

File: Celeste.Mod.mm/Content/Dialog/English.txt (hunk 1 above)
Comment (JaThePlayer) [reply Celeste.Mod.mm/Content/Dialog/English.txt]: Alright, they're removed now

**Pull request merged**